### ISSUE TITLE
Up to 6 seconds observed on demo when there are many integrations - some triggering on overlapping words

### ISSUE THREAD

**Issue by shaharyakir** [RocketChat/Rocket.Chat#4938]

**Comment by MartinSchoeler** [comment]
hey @shaharyakir could you give us more info on what you mean? like some screenshots or saying where the words are overlapping?

**Comment by shaharyakir** [comment]
Hi this issue was opened as per @Sing-Li request.

I think by overlapping he meant that the same keyword triggering the integration could point to more than one integration.

**Comment by graywolf336** [comment]
@shahar3012 and @Sing-Li can you guys provide some step by step so that it can be reproduced

**Comment by shaharyakir** [comment]
Hi, Just write "guggy hello" in the demo. 

https://demo.rocket.chat/channel/general

**Comment by JamieH** [comment]
[Issue can be seen here](https://share.jamiehankins.co.uk/ShareX/2016/11/chrome_2016-11-20_03-43-35.png)

**Comment by mehdiyahyaoui** [comment]
Hello 

I'm Running 0.50.0
Snap
Ubuntu 64

I created the webhook with indication from Doc 

But I have this error message in the Log:

` Exception` in callback of async function: TypeError: Cannot read property 'statusCode' of null   at /snap/rocketchat-server/446/programs/server/packages/rocketchat_integrations.js:1316:30   at runWithEnvironment (packages/meteor.js:1176:24) `

Trigger Work is "guggy" without ! mark
Basically nothing is happening when I type : `guggy "something"  `
